2019/pkg/intcode/intcode.go

```diff
@@ -0,0 +1,154 @@
+package intcode
+
+import (
+	"errors"
+	"fmt"
+)
+
+type opCode int
+
+const (
+	opCodeAdd       opCode = 1
+	opCodeMultiply  opCode = 2
+	opCodeRead      opCode = 3
+	opCodeWrite     opCode = 4
+	opCodeTerminate opCode = 99
+	opCodeUnknown   opCode = -1
+)
+
+var intToOpCode = map[int]opCode{
+	1:  opCodeAdd,
+	2:  opCodeMultiply,
+	3:  opCodeRead,
+	4:  opCodeWrite,
+	99: opCodeTerminate,
+	-1: opCodeUnknown,
+}
+
+var opCodeInstructionCount = map[opCode]int{
+	opCodeAdd:       4,
+	opCodeMultiply:  4,
+	opCodeRead:      2,
+	opCodeWrite:     2,
+	opCodeTerminate: 1,
+}
+
+type parameterMode int
+
+const (
+	// positionMode interprets a parameter as a postion.
+	positionMode parameterMode = 0
```

Comment:
I would call this `absoluteAddress` (https://en.wikipedia.org/wiki/Addressing_mode#Simple_addressing_modes_for_data) and the other `immediateValue`.